src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -53,39 +53,64 @@ export async function getNewData(
             }
             newData.parameters.startTime = isViewAllEvents(newTimeRange) ? undefined : newTimeRange.start
             newData.parameters.endTime = isViewAllEvents(newTimeRange) ? undefined : newTimeRange.end
-
             break
     }
+    let resourceType: telemetry.CloudWatchResourceType = 'logGroup'
+
     if (newData.logGroupInfo.streamName) {
         newData.retrieveLogsFunction = filterLogEventsFromUriComponents
         newData.parameters.streamNameOptions = [newData.logGroupInfo.streamName]
         newData.logGroupInfo.streamName = undefined
+        resourceType = 'logStream'
     }
 
+    telemetryFilter(newData, resourceType, 'escapeHatch')
+
     return newData
 }
 
 export async function changeLogSearchParams(
     registry: LogStreamRegistry,
     param: 'filterPattern' | 'timeFilter'
 ): Promise<void> {
-    //let result: telemetry.Result = 'Succeeded'
     const oldUri = getActiveDocumentUri(registry)
 
     const oldData = registry.getLogData(oldUri)
     if (!oldData) {
+        telemetry.recordCloudwatchlogsFilter({
+            result: 'Failed',
+            source: 'escapeHatch',
+            cloudWatchResourceType: isLogStreamUri(oldUri) ? 'logStream' : 'logGroup',
+            hasTimeFilter: param === 'timeFilter',
+            hasTextFilter: param === 'filterPattern',
+        })
         throw new Error(`cwl: Unable to find data for active URI ${oldUri}`)
     }
     const newData = await getNewData(param, oldData)
 
     if (!newData) {
-        //result = 'Cancelled'
+        telemetry.recordCloudwatchlogsFilter({
+            result: 'Cancelled',
+            source: 'escapeHatch',
+            cloudWatchResourceType: isLogStreamUri(oldUri) ? 'logStream' : 'logGroup',
+            hasTimeFilter: oldData.parameters.startTime || param === 'timeFilter' ? true : false,
```

Comment:
```suggestion
            hasTimeFilter: oldData.parameters.startTime || param === 'timeFilter',
```

Reply:
had to add a `!== undefined` to avoid linting error, but believe it has the same function.

Reply:
Usually if you see these kinds of errors you can do this:
```ts
!!(oldData.parameters.startTime || param === 'timeFilter')
```

This forces the expression into a boolean.